src/Rules/Functions/PrintfHelper.php

```diff
@@ -44,34 +55,83 @@ private function getPlaceholdersCount(string $specifiersPattern, string $format)
 		$matches = Strings::matchAll($format, $pattern, PREG_SET_ORDER);
 
 		if (count($matches) === 0) {
-			return 0;
+			return [];
 		}
 
 		$placeholders = array_filter($matches, static fn (array $match): bool => strlen($match['before']) % 2 === 0);
 
-		if (count($placeholders) === 0) {
-			return 0;
-		}
+		$result = [];
+		$parsedPlaceholders = [];
+		$parameterIdx = 0;
+		$placeholderNumber = 0;
 
-		$maxPositionedNumber = 0;
-		$maxOrdinaryNumber = 0;
 		foreach ($placeholders as $placeholder) {
+			$placeholderNumber++;
+			$showValueSuffix = false;
+
 			if (isset($placeholder['width']) && $placeholder['width'] !== '') {
-				$maxOrdinaryNumber++;
+				$parsedPlaceholders[] = new PrintfPlaceholder(
+					sprintf('"%s" (width)', $placeholder[0]),
+					$parameterIdx++,
+					$placeholderNumber,
+					'strict-int',
+				);
+				$showValueSuffix = true;
 			}
 
 			if (isset($placeholder['precision']) && $placeholder['precision'] !== '') {
-				$maxOrdinaryNumber++;
+				$parsedPlaceholders[] = new PrintfPlaceholder(
+					sprintf('"%s" (precision)', $placeholder[0]),
+					$parameterIdx++,
+					$placeholderNumber,
+					'strict-int',
+				);
+				$showValueSuffix = true;
 			}
 
-			if (isset($placeholder['position']) && $placeholder['position'] !== '') {
-				$maxPositionedNumber = max((int) $placeholder['position'], $maxPositionedNumber);
-			} else {
-				$maxOrdinaryNumber++;
-			}
+			$parsedPlaceholders[] = new PrintfPlaceholder(
+				sprintf('"%s"', $placeholder[0]) . ($showValueSuffix ? ' (value)' : ''),
+				isset($placeholder['position']) && $placeholder['position'] !== ''
+					? $placeholder['position'] - 1
+					: $parameterIdx++,
+				$placeholderNumber,
+				$this->getAcceptingTypeBySpecifier($placeholder['specifier'] ?? ''),
```

Comment:
This says "accepting" but the property it's being passed to is "accepted".

"Accepting" and "accepted" are opposite sides of the equation so please update it to reflect what side is this.